src/app/feed/shared/feed-connection.model.spec.ts

```diff
@@ -30,102 +30,248 @@ describe("Service: FeedConnection", () => {
-    fConnection.setAttrs(
-      this.pluginHandle,
-      1
-    );
+  describe("#handleRemoteJsep", () => {
+    it("should call pluginHandle.handleRemoteJsep", () => {
+      let jsep: any = { id: 13 };
+      this.connection.handleRemoteJsep(jsep);
 
-    fConnection.register("display");
-
-    expect(this.pluginHandle.send).toHaveBeenCalledWith(
-      jasmine.objectContaining({
-        message: {
-          request: "join",
-          room: 1,
-          ptype: "publisher",
-          display: "display"
-        }
-      })
-    );
+      expect(this.pluginHandle.handleRemoteJsep).toHaveBeenCalledWith(
+        jasmine.objectContaining({
+          jsep: jsep
+        })
+      );
+    });
   });
 
-  it("should call pluginHandle.send on call listen", () => {
-    let fConnection: FeedConnection = new FeedConnection();
-    fConnection.setAttrs(
-      this.pluginHandle,
-      1
-    );
+  describe("#sendData", () => {
+    it("should call pluginHandle.data", () => {
+      let data = { id: 1 };
+      this.connection.sendData(data);
 
-    fConnection.listen(5);
-
-    expect(this.pluginHandle.send).toHaveBeenCalledWith(
-      jasmine.objectContaining({
-        message: {
-          request: "join",
-          room: 1,
-          ptype: "listener",
-          feed: 5
-        }
-      })
-    );
+      expect(this.pluginHandle.data).toHaveBeenCalledWith(data);
+    });
   });
 
-  it("should call pluginHandle.handleRemoteJsep when call handleRemoteJsep", () => {
-    let fConnection: FeedConnection = new FeedConnection();
-    fConnection.setAttrs(
-      this.pluginHandle,
-      1
-    );
+  describe("#publish", () => {
+    beforeEach(() => {
+      this.options = jasmine.createSpyObj("options", ["success", "error"]);
+    });
 
-    let jsep: any = { id: 13 };
-    fConnection.handleRemoteJsep(jsep);
+    it("should create a webRTC offer as subscriber", () => {
+      this.connection.publish();
+
+      expect(this.pluginHandle.createOffer).toHaveBeenCalledWith(
+        jasmine.objectContaining({
+          media: {
+            videoRecv: false,
+            audioRecv: false,
+            video: "screen",
+            audioSend: false,
+            data: false
+          },
+          success: jasmine.any(Function),
+          error: jasmine.any(Function)
+        })
+      );
+    });
+
+    it("should create a webRTC offer as main", () => {
+      this.connection.role = "main";
+      this.connection.publish();
+
+      expect(this.pluginHandle.createOffer).toHaveBeenCalledWith(
+        jasmine.objectContaining({
+          media: {
+            videoRecv: false,
+            audioRecv: false,
+            audioSend: true,
+            videoSend: true,
+            data: true
+          },
+          success: jasmine.any(Function),
+          error: jasmine.any(Function)
+        })
+      );
+    });
+
+    it("should call success callback when webRTC offer success", () => {
```

Comment:
I think this test does not belong to here as you're testing the internals of plugin handle. IMHO it makes sense as a pluginHandle unit test, but not as part of `FeedConnection` model.

On the other hand, you're faking the `createOffer` (and calling `options.success()`), so I think that the original code does not get executed.

The same applies to the test below.